### ISSUE TITLE
Decide what JAX settings tabascal owns: precision and matmul are still assigned over the user

### ISSUE THREAD

**Issue by chrisfinlay** [epfl-radio-astro/tabascal#215]
#214 fixed one instance of a general problem: tabascal set a JAX environment variable with `os.environ[...] = ...` rather than `setdefault`, so a user who exported `XLA_PYTHON_CLIENT_PREALLOCATE` had it silently discarded. The same pattern exists for the JAX **config** knobs, which are settable by environment variable too, and `set_precision` overwrites two of them unconditionally.

The general question this issue is for: **which JAX settings are tabascal's to decide, which are the user's, and what happens when they disagree?** The answer is not the same for all of them, which is why this is worth doing deliberately rather than by sweeping `setdefault` over everything.

## 1. `jax_default_matmul_precision` — overwritten, and no config key claims it

[`_run_tabascal_impl.py:532`](https://github.com/epfl-radio-astro/tabascal/blob/main/tabascal/scripts/_run_tabascal_impl.py#L532):

```python
jax.config.update("jax_default_matmul_precision", "highest")
```

Unconditional, and nothing in the config names this setting. Measured:

```console
$ JAX_DEFAULT_MATMUL_PRECISION=float32 python -c "..."
before  : float32     # JAX read the export
after   : highest     # set_precision took it back
```

This is #94's bug exactly — an export honoured until tabascal runs, then discarded, with no message. There is a good reason for the default (the comment above it explains that TF32 matmuls wreck single-precision visibility algebra on Ampere+, reduced chi^2 to ~1e14), but "a sound default" is the argument for `setdefault`, not for assignment. A user deliberately trading accuracy for speed on a machine they understand should be able to.

**Wanted:** default it, do not impose it. If it must be imposed for correctness, say so out loud when the user asked for something else rather than silently winning.

## 2. `jax_enable_x64` — overwritten, but here the config genuinely owns it

Same call site, [line 531](https://github.com/epfl-radio-astro/tabascal/blob/main/tabascal/scripts/_run_tabascal_impl.py#L531). Measured:

```console
$ JAX_ENABLE_X64=1 python -c "... set_precision({}) ..."
  before set_precision: True
  after  set_precision: False
```

A config that says nothing about precision defaults to `single` (`.get("precision", "single")`) and turns off an x64 the user explicitly asked for.

**This one should probably keep winning.** `model.precision` is a scientific choice recorded in the config, reproducible and reviewable, and an environment variable quietly changing what a run computes would be worse than the current behaviour. But the resolution should be *stated*: a run whose config and environment disagree about precision should say which won, in the same place it already prints the precision it is using. Silence is the defect here, not the precedence.

## 3. The save/restore pairs restore JAX's state, not the user's intent

[`rfi_estimate.py:722-723`](https://github.com/epfl-radio-astro/tabascal/blob/main/tabascal/scripts/rfi_estimate.py#L722) and [`sat_search.py:636-637`](https://github.com/epfl-radio-astro/tabascal/blob/main/tabascal/scripts/sat_search.py#L636) both restore `was_x64` / `was_matmul` around a scan. Correct within a process, and worth keeping — but the value restored is whatever `set_precision` last imposed, so the export never gets a look-in on these paths either.

## 4. There is no one place that says what tabascal touches

`XLA_PYTHON_CLIENT_PREALLOCATE` now lives in [`scripts/_device_memory.py`](https://github.com/epfl-radio-astro/tabascal/blob/main/tabascal/scripts/_device_memory.py), which is the shape to copy. The rest is scattered: `set_precision` for two config knobs; `ORBIT_CACHE_DIR`, `TAB_LOSS_TRACE`, `TABASCAL_FORCE_DISTRIBUTED` and `MASTER_ADDR`/`MASTER_PORT` read in four other modules; `CUDA_VISIBLE_DEVICES` pinned in `ci/cscs.yml` and in `tests/conftest.py:20` — the latter already with `setdefault`, which is the habit worth generalising.

## Wanted

- **One module** owning every environment variable and JAX config knob tabascal sets, in the shape `_device_memory.py` established, with the reason for each default written down beside it.
- **A stated policy per setting**: tabascal's to decide, the user's to decide, or the config's — and, where tabascal or the config wins, a message rather than silence.
- **`jax_default_matmul_precision` defaulted rather than assigned**, since nothing else claims it.
- **The precision conflict reported** where the run already prints its precision.
- **Documented** in the GPU-memory section of `docs/usage.md` that #214 added, which is where a reader will look.

## Acceptance

- [ ] No JAX environment variable or config knob is assigned over a user's explicit setting without either a documented reason or a message saying so
- [ ] `JAX_DEFAULT_MATMUL_PRECISION` survives a run that does not need to override it
- [ ] A run whose `model.precision` disagrees with an exported `JAX_ENABLE_X64` says which it used
- [ ] One place lists what tabascal sets and why; `docs/usage.md` points at it
- [ ] The AST guard from #214 (`test_nothing_in_the_package_assigns_it`) extended to the settings that gain a defer-to-the-user policy

## Related

- #94 — the first instance, fixed in #214
- #214 — the fix, and the `_device_memory.py` shape to copy
